presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1117,6 +1120,42 @@ protected Scope visitJoin(Join node, Optional<Scope> scope)
             Scope right = process(node.getRight(), isLateralRelation(node.getRight()) ? Optional.of(left) : scope);
 
             if (criteria instanceof JoinUsing) {
+                if (SystemSessionProperties.isLegacyJoinUsingEnabled(session)) {
+                    List<Identifier> columns = ((JoinUsing) criteria).getColumns();
```

Comment:
I would wrap this a method `legacyAnalyzeJoinUsing`. Just to make it more readable.